### ISSUE TITLE
Allow "salvage" on a "busy" robot.

### ISSUE THREAD

**Issue by kostmo** [swarm-game/swarm#1025]
Apparently the `salvage` command only works on a robot that is `idle`.  But a built robot that mistakenly was sent into an infinite loop is a good use case for salvaging and trying again.

Is there a specific reason why `salvage` should not work on a `busy` robot?

**Comment by xsebek** [comment]
@kostmo yes, It would not be good manners. :smile: 

I think that once we allow `salvage` to work on busy robots, you would agree that `salvage` suddenly has too many responsibilities and should be split into a command that stops robots...

In `halt`! #392

**Comment by xsebek** [comment]
Technically, there is also a _slight issue_ with how `salvage` currently works.

We make the salvaged robot busy again and make it give each inventory item to the salvaging robot.

If you were able to `salvage` (or `halt`) the salvaged robot again then the original salvaging robot would keep waiting the amount of time it expected salvaging would take and not get cancelled.

But hopefully, no items would magically (dis)appear if you did that, so it's not a big problem. :slightly_smiling_face:

**Comment by byorgey** [comment]
Yeah, my gut feeling is that `salvage` on a busy robot is too heavy-handed, and would be a rather blunt instrument: what if a robot crashed, but it's sitting on the same cell as some other, working robots?  Right now, you can send a robot to `salvage` it, confident that you won't disrupt any robots that are actually working.  But if `salvage` could act on busy robots too, I don't know how you would control it.  Maybe because of your playstyle or the kinds of things you've been working on, you haven't had lots of robots on the same cell, but with my automated production facilities this would be a big problem.

I agree that we need something to deal with robots mistakenly sent into infinite loops; I've done that and it's annoying that you can never get them back.  But #392 is my idea for dealing with that.  Since `halt` takes an `actor` as an argument, you can use it a bit more precisely: *e.g.* do a `meetAll` and only call `halt` on the one(s) satisfying some particular criteria, then `salvage`.
